ts_src/transaction_builder.ts

```diff
@@ -438,27 +442,31 @@ export class TransactionBuilder {
 
   private __overMaximumFees(bytes: number): boolean {
     // not all inputs will have .value defined
-    const incoming = this.__INPUTS.reduce((a, x) => a + (x.value! >>> 0), 0);
+    const incoming = this.__INPUTS.reduce(
+      (a, x) =>
+        a + (typeof x.value !== 'undefined' ? BigInt(x.value) : BigInt(0)),
+      BigInt(0),
+    );
 
     // but all outputs do, and if we have any input value
     // we can immediately determine if the outputs are too small
     const outgoing = this.__TX.outs.reduce(
-      (a, x) => a + (x as Output).value,
-      0,
+      (a, x) => a + BigInt((x as Output<TNumber>).value),
+      BigInt(0),
     );
     const fee = incoming - outgoing;
-    const feeRate = fee / bytes;
+    const feeRate = fee / BigInt(bytes);
```

Comment:
I think we should assume that the `fee` (ie, the difference between the input amount and output amount) is representable by a `number`, otherwise we can't properly handle non-integer `feeRate` here since `BigInt` always rounds down. Open to your thoughts on this.
```suggestion
    const feeRate = Number(fee) / bytes;
```